### ISSUE TITLE
Implement PPO with Keras API

### ISSUE THREAD

**Issue by gth828r** [gth828r/ppo-cql#1]
We need an implementation of PPO. Start with using the Keras API.

**Comment by gth828r** [comment]
This video has an excellent description of how PPO works: https://www.youtube.com/watch?v=5P7I-xPq8u8&t=1s&ab_channel=ArxivInsights

As mentioned in that video, a no-frills implementation of PPO is available in PyTorch over at https://github.com/higgsfield/RL-Adventure-2/blob/master/3.ppo.ipynb, which may be a useful reference.

Use those resources to ensure understanding of what is going on.  This issue will also need to be performed in conjunction with #3 and possibly #2, as we won't be able to close this issue out until we have the ability to validate the implementation.

**Comment by gth828r** [comment]
I am making good progress, but I still have some work to do on understanding everything. My current snag is understanding the appropriate way to calculate the entropy bonus over the policy. I have some intuition of what is going on, but I still need to put the pieces together in code.

Once I have the loss function done, I need to implement the top level loop. Then I can go back and work on pulling in gym and testing on cartpole.

**Comment by gth828r** [comment]
The PPO code (in notebook form) has been merged and has been shown to work. PPO will be put into python module form in #7, so this can be closed.
